realsense2_camera/include/realsense_node_factory.h

```diff
@@ -42,10 +45,23 @@ namespace realsense2_camera
 
     const std::vector<stream_index_pair> HID_STREAMS = {GYRO, ACCEL, POSE};
 
+
+    inline void signalHandler(int signum)
+    {
+        ROS_INFO_STREAM(strsignal(signum) << " Signal is received! Terminating RealSense Node...");
+        ros::requestShutdown();
+        ros::shutdown();
+        sleep(5);
+        // exit(signum) is not thread safe and should not be invoked in a multithreaded environment unless all the threads had been joined.
```

Comment:
Any way to check the threads or too much work? Is the commented `exit(signum)` intended to be invoked manually at some point?

Reply:
exit should never be invoked in a multi-threaded application. 

Intel assumes that each camera works in an independent node. Hence their use of exit. Unfortunately we break that assumption by running realsense driver as part of the nodelet manager.  This is forcing me to comment out and remove abort and exit function calls and instead replace them with ros::requestShutdown(); ros::shutDown() etc... (might have to add ros::spinOnce() prior to exiting out of the routine)

There is no way to know about the status of the threads because we are not creating any... it's the nodelet manager that has a pool of threads.